srcts/types/src/shiny/render.d.ts

```diff
@@ -8,21 +8,32 @@ declare function renderContent(el: BindScope, content: string | {
 } | null, where?: WherePosition): void;
 declare function renderHtml(html: string, el: BindScope, dependencies: HtmlDep[], where?: WherePosition): ReturnType<typeof singletonsRenderHtml>;
 declare type HtmlDepVersion = string;
+declare type StylesheetItem = {
+    href: string;
+    [x: string]: string;
+};
+declare type ScriptItem = {
+    src: string;
+    [x: string]: string;
+};
+declare type AttachmentItem = {
+    key: string;
+    href: string;
+    [x: string]: string;
+};
 declare type HtmlDep = {
     name: string;
     version: HtmlDepVersion;
     restyle?: boolean;
     src?: {
         href: string;
     };
-    meta?: string[] | string;
-    stylesheet?: string[] | string;
-    script?: Array<{
-        [key: string]: string;
-    }> | string[] | string | {
-        [key: string]: string;
+    meta?: {
+        [x: string]: string;
```

Comment:
The incoming format is something like:

```ts
{"x": "xval", "y": "yval"}
```

Which gets translated to:

```html
<meta name="x", content="xval">
<meta name="y", content="yval">
```

So we can't really make it more strict.

Note that the `MetaItem[]` format is different. It's normalized to have a structure similar to `ScriptItem[]` and `StylesheetItem[]`, and it looks like this:

```ts
[{name: "x", content: "xval"}, {name: "y", content: "yval"}]
```

The benefits are (A) that it's consistent with the other types of structures, and (B) it also supports other `<meta>` attributes, in case they're wanted. The normalized format is what prism generates, so it could hypothetically make use of the other attributes.